### PR TITLE
So many code changes + documentation

### PR DESCRIPTION
**Pull request by zorg93**
Reworked Camera/Gameplay Camera using their Matrices to replace some of the maths
Added EntityBone/EntityBoneCollection
Cleaned up MemoryPool access
Added loads of documentation

### CODE REVIEW

**Comment by crosire** [comment]
So. Many. Changes. :smile:

**Comment by zorg93** [comment]
Most of the changes is just documentation

**Comment by zorg93** [comment]
Decided to add IExistable and IDeletable, idea being you can create a list of IDeletables to enable easy clean up

File: source/scripting/Entity.cs [hunk 1]
```diff
@@ -37,37 +37,93 @@ public IntPtr MemoryAddress
...
-		/// The health from 0 - 100.
+		/// The health in int.
 		/// </value>
+		/// <remarks>if you need to get or set the value strictly, use <see cref="HealthFloat"/> instead.</remarks>
 		public int Health
 		{
 			get
 			{
-				return Function.Call<int>(Hash.GET_ENTITY_HEALTH, Handle) - 100;
+				return Function.Call<int>(Hash.GET_ENTITY_HEALTH, Handle);
```
Comment (crosire) [inline source/scripting/Entity.cs]: Not sure this is a good idea. Health below 100 does not seem to be used on normal entities, so it's really confusing.

Comment (zorg93) [inline source/scripting/Entity.cs]: I agree, this was a pull request into mine, I just never realised this changed

**Comment by Fireboyd78** [comment]
I'm honestly surprised _existable_ is even a word. Lol.

File: source/scripting/WeaponComponent.cs [hunk 2]
```diff
@@ -3,6 +3,25 @@
 
 namespace GTA
 {
+	public enum ComponentAttachmentPoint : uint
```
Comment (Fireboyd78) [inline source/scripting/WeaponComponent.cs]: This enum is missing the `u` suffix for the hashes, which seems to be the convention for hash enums.

Comment (zorg93) [inline source/scripting/WeaponComponent.cs]: fixed it especially for you :)

**Comment by kagikn** [comment]
Should we merge this PR? @zorg93 will be busy for the time being though.

**Comment by zorg93** [comment]
@kagikn sorry, I've kind of been busy with a much bigger project recently :(

**Comment by Fireboyd78** [comment]
Welp, GitHub lied. I approved changes made to a _single_ file, not the whole damn thing. 👎

File: source/scripting/Game.cs [hunk 3]
```diff
@@ -33,6 +33,8 @@ public enum GameVersion
 		v1_0_757_4_NoSteam,
 		v1_0_791_2_Steam,
 		v1_0_791_2_NoSteam,
+        v1_0_877_1_Steam,
```
Comment (Fireboyd78) [inline source/scripting/Game.cs]: Spaces2Tabs, my friend.

Comment (zorg93) [inline source/scripting/Game.cs]: feck

**Comment by kagikn** [comment]
The entity max health offset has been moved to 0x2A0 in v1.0.877.1. http://www.unknowncheats.me/forum/grand-theft-auto-v/144028-reversal-thread-32.html

**Comment by kagikn** [comment]
Damn, euphoria signatures must be screwed up again.

**Comment by zorg93** [comment]
more than likely its euphoria offsets that need changing

**Comment by zorg93** [comment]
the signature is bad, damn. Found it again using a different sig, but still need some testing etc

**Comment by crosire** [comment]
I'll go ahead and merge this now so the two branches again are even.

**Pull request merged**